lib/statesman/adapters/active_record_queries.rb

```diff
@@ -1,51 +1,122 @@
 module Statesman
   module Adapters
     module ActiveRecordQueries
+      def self.check_missing_methods!(base)
+        missing_methods = %i[transition_class initial_state].
+          reject { |_method| base.respond_to?(:method) }
+        return if missing_methods.none?
+
+        raise NotImplementedError,
+              "#{missing_methods.join(', ')} method(s) should be defined on " \
+              "the model. Alternatively, use the new form of `extend " \
+              "Statesman::Adapters::ActiveRecordQueries[" \
+              "transition_class: MyTransition, " \
+              "initial_state: :some_state]`"
+      end
+
       def self.included(base)
-        base.extend(ClassMethods)
+        check_missing_methods!(base)
+
+        base.include(
+          ClassMethods.new(
+            transition_class: base.transition_class,
+            initial_state: base.initial_state,
+            most_recent_transition_alias: base.try(:most_recent_transition_alias),
+            transition_name: base.try(:transition_name),
+          ),
+        )
       end
 
-      module ClassMethods
-        def in_state(*states)
-          states = states.flatten.map(&:to_s)
+      def self.[](**args)
+        ClassMethods.new(**args)
+      end
+
+      class ClassMethods < Module
+        def initialize(**args)
+          @args = args
+        end
+
+        def included(base)
+          ensure_inheritance(base)
```

Comment:
Are there any docs to how the inheritance should be setup? 

I have the following setup:

```
    rails (~> 5.2.4)
    enumerize (2.3.1)
    statesman (7.2.0)
```


```ruby
class Candidate < ApplicationRecord
  extend Enumerize
  ...
  include Statesman::Adapters::ActiveRecordQueries[
    initial_state: :incomplete,
    transition_class: CandidateTransition
  ]
  has_many :candidate_transitions, inverse_of: :candidate, dependent: :destroy
  ...
  enumerize :state, in: CandidateStateMachine.states, default: :incomplete, predicates: true
  ...
end
```

```ruby
class Worker < Candidate
  ...
end
```

Upon instantiating a `Worker`, a stack error occurs:

```
     # /home/travis/.rvm/gems/ruby-2.6.4/gems/statesman-7.2.0/lib/statesman/adapters/active_record_queries.rb:60:in `block in ensure_inheritance'
     # /home/travis/.rvm/gems/ruby-2.6.4/gems/enumerize-2.3.1/lib/enumerize/base.rb:37:in `inherited'
     # /home/travis/.rvm/gems/ruby-2.6.4/gems/statesman-7.2.0/lib/statesman/adapters/active_record_queries.rb:60:in `call'
     # /home/travis/.rvm/gems/ruby-2.6.4/gems/statesman-7.2.0/lib/statesman/adapters/active_record_queries.rb:60:in `block in ensure_inheritance'
     # /home/travis/.rvm/gems/ruby-2.6.4/gems/enumerize-2.3.1/lib/enumerize/base.rb:37:in `inherited'
     # /home/travis/.rvm/gems/ruby-2.6.4/gems/statesman-7.2.0/lib/statesman/adapters/active_record_queries.rb:60:in `call'
     # /home/travis/.rvm/gems/ruby-2.6.4/gems/statesman-7.2.0/lib/statesman/adapters/active_record_queries.rb:60:in `block in ensure_inheritance'
     # /home/travis/.rvm/gems/ruby-2.6.4/gems/enumerize-2.3.1/lib/enumerize/base.rb:37:in `inherited'
     # /home/travis/.rvm/gems/ruby-2.6.4/gems/statesman-7.2.0/lib/statesman/adapters/active_record_queries.rb:60:in `call'
     # /home/travis/.rvm/gems/ruby-2.6.4/gems/statesman-7.2.0/lib/statesman/adapters/active_record_queries.rb:60:in `block in ensure_inheritance'
     # /home/travis/.rvm/gems/ruby-2.6.4/gems/enumerize-2.3.1/lib/enumerize/base.rb:37:in `inherited'
     # /home/travis/.rvm/gems/ruby-2.6.4/gems/statesman-7.2.0/lib/statesman/adapters/active_record_queries.rb:60:in `call'
     # /home/travis/.rvm/gems/ruby-2.6.4/gems/statesman-7.2.0/lib/statesman/adapters/active_record_queries.rb:60:in `block in ensure_inheritance'
     # /home/travis/.rvm/gems/ruby-2.6.4/gems/enumerize-2.3.1/lib/enumerize/base.rb:37:in `inherited'
     # ./app/models/worker.rb:1:in `<top (required)>'
```

Removing the `ensure_inheritance(base)` line from ` lib/statesman/adapters/active_record_queries.rb` allows me to instantiate the `Worker` class as expected, and I can call the statesman methods on the model:

```
Worker.last.history
  Worker Load (0.5ms)  SELECT  "candidates".* FROM "candidates" ORDER BY "candidates"."id" DESC LIMIT $1  [["LIMIT", 1]]
  CandidateTransition Load (3.8ms)  SELECT "candidate_transitions".* FROM "candidate_transitions" WHERE "candidate_transitions"."candidate_id" = $1 ORDER BY "candidate_transitions"."sort_key" ASC  [["candidate_id", xxxx]]
=> [#<CandidateTransition:0x000000010dbaee20
  id: xxxx,
  candidate_id: xxxx,
  to_state: "active",
  metadata: {},
  sort_key: 10,
  created_at: Mon, 03 Aug 2020 11:05:28 UTC +00:00,
  created_by_id: xxxx,
  most_recent: true,
  updated_at: Mon, 03 Aug 2020 11:05:28 UTC +00:00>]
```

Reply:
The [issue above](https://github.com/gocardless/statesman/pull/373#discussion_r465877454) is still occurring using:

```
    rails (7.0.4.3)
    enumerize (2.6.1)
    statesman (10.2.2)
```

I've opened https://github.com/gocardless/statesman/pull/511